api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -118,4 +118,17 @@ public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
    * @return this for method chaining
    */
   ExpireSnapshots cleanExpiredFiles(boolean clean);
+
+  /**
+   * Allows expiration of unreachable table layout metadata, such as partition specs as part of the
+   * operation.
+   *
+   * @param clean setting this to true will remove table layout metadata that is no longer reachable
+   *     by any snapshot
```

Comment:
This is a bit verbose. I'd recommend being more direct and not making it a full sentence. Something like:
> remove unused partition specs, schemas, or other metadata when true

The same could apply to the method description:
> Enable cleaning up unused partition specs, schemas, or other metadata.